lib/fs_utils/source_file.js

```diff
@@ -134,6 +134,11 @@ class SourceFile {
     this.removed = false;
     this.disposed = false;
     this.compile = makeCompiler(path, this, linters, compilers, wrap, jsWrap, depCompilers);
+
+    if (type === 'static') {
+      this.targetExtension = compilers[0].targetExtension;
+    }
```

Comment:
this is needed so that we later know which extension to use for the file when writing